routing/router.go

```diff
@@ -288,6 +289,12 @@ type Config struct {
 	// we need in order to properly maintain the channel graph.
 	ChainView chainview.FilteredChainView
 
+	// Notifier is a reference to the ChainNotifier, used to grab
+	// the latest blocks if the router is missing any.
+	Notifier chainntnfs.ChainNotifier
+
+	BlockNotifier *chainntnfs.BlockEpochEvent
```

Comment:
Generally, we don't want to have members that are used only for tests to be in the config.  I think this can be removed.  The alternative approach here would be to modify `lntest/mock`s `ChainNotifier` to handle the `RegisterBlockEpochNtfn` backlog appropriately.  Alternatively, a router-specific mock could be used, but since we want to have a specific place for mocks if possible, I'd like to avoid that.